overlord/fdestate/backend/reseal.go

```diff
@@ -60,7 +60,10 @@ func MockSecbootBuildPCRProtectionProfile(f func(modelParams []*secboot.SealKeyM
 	}
 }
 
-type StateUpdater func(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error
+type FDEStateManager interface {
+	Update(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error
+	Get(role string, containerRole string) (bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte, err error)
```

Comment:
not something we can do much about but is indeed not great that both things are called role, we need to be very careful

Reply:
We should use types at least. Again, this is related to the FIXMEs on calls to Update.